src/wp-includes/media.php

```diff
@@ -275,26 +275,36 @@ function image_downsize( $id, $size = 'medium' ) {
  * Register a new image size.
  *
  * @since 2.9.0
+ * @since 6.1.0 Add the $output_mimes parameter.
  *
  * @global array $_wp_additional_image_sizes Associative array of additional image sizes.
  *
- * @param string     $name   Image size identifier.
- * @param int        $width  Optional. Image width in pixels. Default 0.
- * @param int        $height Optional. Image height in pixels. Default 0.
- * @param bool|array $crop   Optional. Image cropping behavior. If false, the image will be scaled (default),
- *                           If true, image will be cropped to the specified dimensions using center positions.
- *                           If an array, the image will be cropped using the array to specify the crop location.
- *                           Array values must be in the format: array( x_crop_position, y_crop_position ) where:
- *                               - x_crop_position accepts: 'left', 'center', or 'right'.
- *                               - y_crop_position accepts: 'top', 'center', or 'bottom'.
+ * @param string     $name             Image size identifier.
+ * @param int        $width            Optional. Image width in pixels. Default 0.
+ * @param int        $height           Optional. Image height in pixels. Default 0.
+ * @param bool|array $crop             Optional. Image cropping behavior. If false, the image will be scaled (default),
+ *                                     If true, image will be cropped to the specified dimensions using center positions.
+ *                                     If an array, the image will be cropped using the array to specify the crop location.
+ *                                     Array values must be in the format: array( x_crop_position, y_crop_position ) where:
+ *                                         - x_crop_position accepts: 'left', 'center', or 'right'.
+ *                                         - y_crop_position accepts: 'top', 'center', or 'bottom'.
+ * @param bool       $additional_mimes Optional. Whether to output secondary mimes for this image size. Default is null
+ *                                     but may change to true in future versions. Sizes should be registered with false
+ *                                     only if they are not intended for use in a front-end context.
  */
-function add_image_size( $name, $width = 0, $height = 0, $crop = false ) {
+function add_image_size( $name, $width = 0, $height = 0, $crop = false, $additional_mimes = null ) {
 	global $_wp_additional_image_sizes;
 
+	// Trigger a warning if the size was registered without providing an `additional_mimes` parameter.
+	if ( null === $additional_mimes ) {
+		_doing_it_wrong( __FUNCTION__, __( 'You should provide a value for the `additional_mimes` parameter.' ), '6.1.0' );
+	}
```

Comment:
We also need to set an actual boolean value here if no value is passed. This is where we'll initially set it to `false`, but it could later change.

Also, please use real HTML instead of Markdown :)
```suggestion
	if ( null === $additional_mimes ) {
		_doing_it_wrong(
			__FUNCTION__,
			sprintf(
				/* translators: %s: additional_mimes variable */
				__( 'You should provide a boolean value for the %s parameter.' ),
				'<code>$additional_mimes</code>'
			),
			'6.1.0'
		);
		$additional_mimes = false;
	}
```